### ISSUE TITLE
Test failure: UnexpectedExceptionDuringFrameProcessingLoggedAWarning

### ISSUE THREAD

**Issue by ryanbrandenburg** [aspnet/KestrelHttpServer#2800]
This test [fails](http://aspnetci/viewLog.html?buildId=526890&buildTypeId=XPlat_Windows_Win2008r2_Universe) occasionally with the following error:
```
System.InvalidOperationException : Sequence contains more than one element
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http2ConnectionTests.UnexpectedExceptionDuringFrameProcessingLoggedAWarning() in /_/test/Kestrel.Core.Tests/Http2ConnectionTests.cs:line 3452
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

CC @muratg

This issue was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526615&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
UnexpectedExceptionDuringFrameProcessingLoggedAWarning [failed with about the same error](http://aspnetci/viewLog.html?buildId=526488&buildTypeId=XPlat_Windows_Win2008r2_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
UnexpectedExceptionDuringFrameProcessingLoggedAWarning [failed with about the same error](http://aspnetci/viewLog.html?buildId=526494&buildTypeId=XPlat_Windows_Win8_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526477&buildTypeId=XPlat_Windows_Win7_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
UnexpectedExceptionDuringFrameProcessingLoggedAWarning [failed with about the same error](http://aspnetci/viewLog.html?buildId=526486&buildTypeId=XPlat_HighSierraOsxUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
UnexpectedExceptionDuringFrameProcessingLoggedAWarning [failed with about the same error](http://aspnetci/viewLog.html?buildId=526492&buildTypeId=XPlat_Mac_MacUniverseCoreCLR).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526508&buildTypeId=Releases_22xPublic_Win2012).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526475&buildTypeId=XPlat_Windows_Win10Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526453&buildTypeId=XPlat_Windows_Win2008r2_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526337&buildTypeId=Releases_22xPublic_UbuntuUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
UnexpectedExceptionDuringFrameProcessingLoggedAWarning [failed with about the same error](http://aspnetci/viewLog.html?buildId=526341&buildTypeId=Releases_22xPublic_HighSierraOsxUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526339&buildTypeId=Releases_22xPublic_OsxUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526280&buildTypeId=XPlat_Windows_Win8_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526278&buildTypeId=XPlat_Mac_MacUniverseCoreCLR).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526263&buildTypeId=XPlat_Windows_Win7_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
UnexpectedExceptionDuringFrameProcessingLoggedAWarning [failed with about the same error](http://aspnetci/viewLog.html?buildId=526305&buildTypeId=Releases_22xPublic_UbuntuUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
UnexpectedExceptionDuringFrameProcessingLoggedAWarning [failed with about the same error](http://aspnetci/viewLog.html?buildId=526276&buildTypeId=XPlat_Linux_UbuntuUniverseCoreCLR).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
UnexpectedExceptionDuringFrameProcessingLoggedAWarning [failed with about the same error](http://aspnetci/viewLog.html?buildId=526261&buildTypeId=XPlat_Windows_Win10Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526248&buildTypeId=Releases_22xPublic_OsxUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526224&buildTypeId=XPlat_Mac_MacUniverseCoreCLR).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
UnexpectedExceptionDuringFrameProcessingLoggedAWarning [failed with about the same error](http://aspnetci/viewLog.html?buildId=526220&buildTypeId=XPlat_Windows_Win2008r2_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
IOExceptionDuringFrameProcessingLoggedAsInfo [failed with about the same error](http://aspnetci/viewLog.html?buildId=526218&buildTypeId=XPlat_HighSierraOsxUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
UnexpectedExceptionDuringFrameProcessingLoggedAWarning [failed with about the same error](http://aspnetci/viewLog.html?buildId=527413&buildTypeId=XPlat_Linux_UbuntuUniverseCoreCLR) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by halter73** [comment]
Fixed by #2795
